### PR TITLE
Feature/association screen

### PR DESCRIPTION
**Pull request by ntissieres**
Adding the managing screen which will decide which page to display between AssociationMainScreen, AssociationDetails and AssociationSearch

### CODE REVIEW

File: app/src/main/java/com/github/swent/echo/compose/association/AssociationSearchBottomSheet.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import androidx.compose.runtime.Composable
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AssociationSearchBottomSheet(
+    onFullyExtended: () -> Unit,
+    onDismiss: () -> Unit,
+    searchEntryCallback: (String) -> Unit,
+    searched: String
+) {}
```
Comment (Greenade) [inline app/src/main/java/com/github/swent/echo/compose/association/AssociationSearchBottomSheet.kt:13]: another placeholder, seems like a lot of work to me for other branches, maybe you could request help for all the work.

**Review by Greenade — CHANGES_REQUESTED** [review]
Thanks for your work ! Seems to me like there are a lot of placeholders, maybe you could ask for help on those as it looks like there's a lot of work to do left. Other than that everything looks great 💯 Just change the name of the variable I mentioned above and we're good 😄

File: app/src/main/java/com/github/swent/echo/compose/association/AssociationDetails.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import com.github.swent.echo.data.model.Event
+
+@Composable
+fun AssociationDetails(
+    follow: (Association) -> Unit,
+    association: Association,
+    isFollowed: Boolean,
+    events: List<Event>
+) {
+    Text("Association Details", modifier = Modifier.testTag("association_details"))
```
[the remark this answers is not in the document]
Comment (ntissieres) [reply app/src/main/java/com/github/swent/echo/compose/association/AssociationDetails.kt:17]: the next PR arrive juste after this merge :)

File: app/src/main/java/com/github/swent/echo/compose/association/AssociationSearchBottomSheet.kt (hunk 1 above)
Comment (ntissieres) [reply app/src/main/java/com/github/swent/echo/compose/association/AssociationSearchBottomSheet.kt:13]: I already finished everything on an intermediate PR ! Just pushing one by one and adding the tests ;)

**Review by Greenade — DISMISSED** [review]

**Review by Greenade — APPROVED** [review]
Everything is good now 👍

**Pull request merged**